atlasdb-api/src/main/java/com/palantir/atlasdb/namespacedeleter/NamespaceDeleter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.namespacedeleter;
+
+import java.io.Closeable;
+
+/**
+ * Allows for deleting all data from a namespace (e.g., dropping the keyspace for a Cassandra KVS) programmatically,
```

Comment:
the reason we didn't say 'deleting a namespace' is because of timelock, which we don't flush (but that's ok: timelock doesn't store a lot of information)